dbt/tests/adapter/unit_testing/test_case_insensitivity.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+from dbt.tests.util import run_dbt
+from dbt.tests.fixtures.project import *
+
+
+my_model_sql = """
+select
+    tested_column from {{ ref('my_upstream_model')}}
+"""
+
+my_upstream_model_sql = """
+select 1 as tested_column
+"""
+
+test_my_model_yml = """
+unit_tests:
+  - name: test_my_model
+    model: my_model
+    given:
+      - input: ref('my_upstream_model')
+        rows:
+          - {tested_column: 1}
+          - {TESTED_COLUMN: 2}
+          - {tested_colUmn: 3}
+    expect:
+      rows:
+          - {tested_column: 1}
+          - {TESTED_COLUMN: 2}
+          - {tested_colUmn: 3}
+"""
+
+
+class BaseUnitTestCaseInsensivity:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "my_model.sql": my_model_sql,
+            "my_upstream_model.sql": my_upstream_model_sql,
+            "unit_tests.yml": test_my_model_yml,
+        }
+
+    def test_case_insensitivity(self, project):
+        results = run_dbt(["run"])
+        assert len(results) == 2
+
+        results = run_dbt(["test"])
+
+
+class TestPosgresUnitTestCaseInsensitivity(BaseUnitTestCaseInsensivity):
```

Comment:
There shouldn't be any actual test cases in `dbt-tests-adapter`. This should be implemented in the adapter implementation (e.g. `dbt-postgres`). One of the issues with creating a test case like this is that it cannot be imported without `pytest` running it. So it either doesn't get imported, gets imported and not used (accept to run it as-is), or gets imported because it needs to be modified to run. In the last case, the imported version would likely fail since the point is that it needs to be modified.

Reply:
Totally agree! Discussed offline: I've left these here as part of this work because I understand dbt-postgres is not yet set up to be running its integration tests, and because there will be a migration of postgres tests defined in dbt-adapters into dbt-postgres and moving these can be part of that more unified lift.